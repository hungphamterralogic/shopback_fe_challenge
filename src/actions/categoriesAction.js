import { createAction } from 'redux-actions';

export const initCategories = createAction('CATEGORIES_INIT');
export const selectCategory = createAction('CATEGORY_SELECT');

export default this;
