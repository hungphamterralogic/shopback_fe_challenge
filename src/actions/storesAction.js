import { createAction } from 'redux-actions';

export const initStores = createAction('STORES_INIT');
export const selectStore = createAction('STORE_SELECT');

export default this;
