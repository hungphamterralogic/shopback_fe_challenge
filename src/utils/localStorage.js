const setValueIntoKey = (key, value) => localStorage.setItem(key, value);

const getValueFromKey = key => localStorage.getItem(key);

module.exports = {
  setValueIntoKey,
  getValueFromKey
};
