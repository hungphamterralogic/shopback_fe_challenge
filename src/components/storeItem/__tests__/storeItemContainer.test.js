import React from 'react';

import { genInitialMockStore, renderWithStore } from 'test/reduxComponentTestHelpers';

import * as storesAction from 'actions/storesAction';
import localStorage from 'utils/localStorage';

import StoreItemContainer from '../storeItemContainer';

describe('StoreItemContainer', () => {
  const item = {
    id: 'idTest',
    logoUrl: 'logoUrlTest',
    displayText: 'displayTextTest'
  };

  test('display properly', () => {
    const store = genInitialMockStore();
    const renderer = renderWithStore(<StoreItemContainer item={item} />, store);

    expect(renderer.toJSON()).toMatchSnapshot();
  });

  test('on click', () => {
    const selectStoreMock = jest.spyOn(storesAction, 'selectStore');
    const setValueIntoKeyMock = jest.spyOn(localStorage, 'setValueIntoKey');

    const store = genInitialMockStore();
    const renderer = renderWithStore(<StoreItemContainer item={item} />, store);

    (function clickOnStoreItem() {
      const storeItem = renderer.root.find(el => el.type === 'div');
      storeItem.props.onClick();
    })();

    expect(selectStoreMock.mock.calls.length).toBe(1);
    expect(selectStoreMock).toBeCalledWith(item.id);

    expect(setValueIntoKeyMock.mock.calls.length).toBe(1);
    expect(setValueIntoKeyMock).toBeCalledWith('store', item.id);
  });
});
