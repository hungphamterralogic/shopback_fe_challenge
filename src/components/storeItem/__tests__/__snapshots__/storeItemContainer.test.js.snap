// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreItemContainer display properly 1`] = `
<div
  className="StoreItem-storeItem-0-1-1"
  onClick={[Function]}
>
  <img
    alt="icon"
    className="StoreItem-icon-0-1-3"
    src="logoUrlTest"
  />
  <span
    className="StoreItem-name-0-1-4"
  >
    displayTextTest
  </span>
</div>
`;
